include/nbl/video/CThreadSafeGPUQueueAdapter.h

```diff
@@ -0,0 +1,49 @@
+#ifndef __IRR_GPU_QUEUE_THREADSAFE_ADAPTER_H_INCLUDED__
+#define __IRR_GPU_QUEUE_THREADSAFE_ADAPTER_H_INCLUDED__
+#include "IGPUQueue.h"
+
+namespace nbl::video
+{
+    class CThreadSafeGPUQueueAdapter : public IGPUQueue
+    {
+    protected:
+        nbl::core::smart_refctd_ptr<IGPUQueue> originalQueue = nullptr;
+        std::mutex m;
+    public:
+        CThreadSafeGPUQueueAdapter(nbl::core::smart_refctd_ptr<IGPUQueue>&& originalQueue) : IGPUQueue(nullptr, originalQueue->getFamilyIndex(), originalQueue->getFlags(), originalQueue->getPriority()), originalQueue(originalQueue) {}
+        CThreadSafeGPUQueueAdapter(CThreadSafeGPUQueueAdapter&& other) : IGPUQueue(nullptr, other.originalQueue->getFamilyIndex(), other.originalQueue->getFlags(), other.originalQueue->getPriority()), originalQueue(other.originalQueue) {}
+        
+        CThreadSafeGPUQueueAdapter(const CThreadSafeGPUQueueAdapter& other) : IGPUQueue(nullptr, other.originalQueue->getFamilyIndex(), other.originalQueue->getFlags(), other.originalQueue->getPriority()), originalQueue(other.originalQueue) {}
+        CThreadSafeGPUQueueAdapter(const nbl::core::smart_refctd_ptr<IGPUQueue>& originalQueue) : IGPUQueue(nullptr, originalQueue->getFamilyIndex(), originalQueue->getFlags(), originalQueue->getPriority()), originalQueue(originalQueue) {}
+        
+        CThreadSafeGPUQueueAdapter() : IGPUQueue(nullptr, 0, E_CREATE_FLAGS::ECF_PROTECTED_BIT, 0.f) {};
+
+        CThreadSafeGPUQueueAdapter& operator=(CThreadSafeGPUQueueAdapter&& other)
+        {
+            originalQueue = other.originalQueue;
+            return *this;
+        }
+        CThreadSafeGPUQueueAdapter& operator=(nbl::core::smart_refctd_ptr<IGPUQueue>&& other)
+        {
+            originalQueue = other;
+            return *this;
+        }
```

Comment:
this makes no sense, `IGPUQueue` is a "Total Interface"

Reply:
same thing as with ctors

Reply:
resolved